### ISSUE TITLE
Closure incorrectly tries to move instead of borrow in `ref` patterns within `if let`

### ISSUE THREAD

**Issue by Veykril** [rust-lang/rust-analyzer#18201]
```rs
struct NotCopy;
struct S(NotCopy);
impl S {
    fn f(&mut self) {
        || {
            if let ref mut cb = self.0 {
            }
        };
    }
}


```
This incorrectly reports `cannot move `NotCopy` out of reference`. This closure is not actually moving the `0` field out.
